python/ray/experimental/data/read_api.py

```diff
@@ -253,7 +254,12 @@ def read_binary_files(
     Returns:
         Dataset holding Arrow records read from the specified paths.
     """
-    raise NotImplementedError  # P0
+    dataset = from_items(paths, parallelism=parallelism)
+    return dataset.map(
```

Comment:
Can we pack the files into tasks so that the number of output blocks == num_parallelism?

Current the number of output blocks == num_files.